imap_processing/ialirt/l0/process_swapi.py

```diff
@@ -88,47 +88,32 @@ def optimize_pseudo_parameters(
 
     Returns
     -------
-    solution_dict : dict
-        Dictionary containing the optimized speed, density, and temperature values for
-        each sweep included in the input count_rates array.
+    pseudo_speed : float
+        Pseudo speed.
+    pseudo_density : float
+        Pseudo density.
+    pseudo_temperature : float
+        Pseudo temperature.
     """
-    solution_dict = {  # type: ignore
-        "pseudo_speed": [],
-        "pseudo_density": [],
-        "pseudo_temperature": [],
-    }
-
-    for sweep in np.arange(count_rates.shape[0]):
-        current_sweep_count_rates = count_rates[sweep, :]
-        current_sweep_count_rate_errors = count_rate_error[sweep, :]
-        # Find the max count rate, and use the 5 points surrounding it
-        max_index = np.argmax(current_sweep_count_rates)
-        initial_speed_guess = np.sqrt(energy_passbands[max_index]) * Consts.speed_coeff
-        initial_param_guess = np.array(
-            [
-                initial_speed_guess,
-                5 * (400 / initial_speed_guess) ** 2,
-                60000 * (initial_speed_guess / 400) ** 2,
-            ]
-        )
-        sol = curve_fit(
-            f=count_rate,
-            xdata=energy_passbands.take(
-                range(max_index - 3, max_index + 3), mode="clip"
-            ),
-            ydata=current_sweep_count_rates.take(
-                range(max_index - 3, max_index + 3), mode="clip"
-            ),
-            sigma=current_sweep_count_rate_errors.take(
-                range(max_index - 3, max_index + 3), mode="clip"
-            ),
-            p0=initial_param_guess,
-        )
-        solution_dict["pseudo_speed"].append(sol[0][0])
-        solution_dict["pseudo_density"].append(sol[0][1])
-        solution_dict["pseudo_temperature"].append(sol[0][2])
+    # Find the max count rate, and use the 5 points surrounding it
+    max_index = np.argmax(count_rates)
+    initial_speed_guess = np.sqrt(energy_passbands[max_index]) * Consts.speed_coeff
+    initial_param_guess = np.array(
+        [
+            initial_speed_guess,
+            5 * (400 / initial_speed_guess) ** 2,
+            60000 * (initial_speed_guess / 400) ** 2,
+        ]
+    )
+    sol = curve_fit(
+        f=count_rate,
+        xdata=energy_passbands.take(range(max_index - 3, max_index + 3), mode="clip"),
+        ydata=count_rates.take(range(max_index - 3, max_index + 3), mode="clip"),
+        sigma=count_rate_error.take(range(max_index - 3, max_index + 3), mode="clip"),
+        p0=initial_param_guess,
+    )
 
-    return solution_dict
+    return sol[0][0], sol[0][1], sol[0][2]
```

Comment:
Can you just return the first solution instead of unpacking everything individually?
```suggestion
    return sol[0]
```

Reply:
Yes. Nice idea.